reduce the esh release version to 0.9.0 to make sure that snapshots are taken

kaikreuzer opened the pull request:
The version 0.10.0.b1 was badly chosen - it should have been 0.10.0-b1 instead.
According to Maven versioning, 0.10.0.b1 is a release and thus considered NEWER than any snapshot.
As we include ESH release + snapshot repos in the openHAB snapshot build, this means that 0.10.0-SNAPSHOTS are now ignored and 0.10.0.b1 is used instead. This leads to issues [like this one](https://github.com/openhab/openhab2-addons/pull/3078#issuecomment-359721434).

Signed-off-by: Kai Kreuzer <kai@openhab.org>

lolodomo commented:
I am lost.
I have to test the REST API version in the Android app. What was and what is the pattern for this version ? Should I consider x.y.z.xxxxx and x.y.z-xxxxxx ?

kaikreuzer commented:
@lolodomo Nothing changes on the REST API through this - it will always show the OSGi version of ESH, so normally a timestamp (with the only exception of 0.10.0.b1 for the 2.2.0 release).

This PR here only addressed issues of the Maven build.
